drivers/dac/Kconfig.stm32

```diff
@@ -0,0 +1,35 @@
+# DAC configuration options
+
+# Copyright (c) 2020 Libre Solar Technologies GmbH
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig DAC_STM32
+	bool "STM32 DAC driver"
+	depends on SOC_FAMILY_STM32
+	help
+	  Enable the driver implementation for the stm32xx DAC
+
+if DAC_STM32
+
+config DAC_1
+	bool "STM32 DAC 1"
+	help
+	  Enable DAC 1. Each STM32 DAC has 2 channels.
+
+config DAC_2
+	bool "STM32 DAC 2"
+	help
+	  Enable DAC 2. Each STM32 DAC has 2 channels.
+
+config DAC_3
+	bool "STM32 DAC 3"
+	help
+	  Enable DAC 3. Each STM32 DAC has 2 channels.
+
+config DAC_4
+	bool "STM32 DAC 4"
+	help
+	  Enable DAC 4. Each STM32 DAC has 2 channels.
+
```

Comment:
It would be nice if we could just enable the instances based on their `status` property in the device tree.

Reply:
Seems like a good idea. How can I access DTS properties in Kconfig? Is there an example from other drivers? Quickly looked at some other drivers, but didn't find one that implements this check.

Reply:
@henrikbrixandersen, the DT_INST_ solution is not in use in all STM32 peripheral drivers (and I'm personally not in favor of this solution). I'm awaiting https://github.com/zephyrproject-rtos/zephyr/issues/19285 to provide a sustainable solution to the peripheral instance activation issue. 
So this method is fine for now.

Also, CONFIG_DAC_X are needed anyway for pinmux definition.

Reply:
@martinjaeger Try grepping for `DT_INST_` under `drivers`. Many drivers use this approach.